File: Coding_Competency/LeetCode/50_Pow.py
class Solution(object):
    def myPow(self, x, n):
        factor = x
        if (n > 0):
            while n > 1:
                x *= factor
                n -= 1
        else:
            while n <= 0:
                x /= factor
                n += 1
        return x


print(Solution().myPow(2.1, 3))

File: Coding_Competency/LeetCode/50_Pow.c
#include <stdio.h>

double myPow(double x, int n){
    double factor = x;
    if (n > 0) {
        while (n > 1) {
            x *= factor;
            n--;
        }
    } else {
        while (n <= 0) {
            x /= factor;
            n++;
        }
    }
    return x;
}

int main() {
    printf("%lf", myPow(2.1, 3));
}